projects/cdk/pipes/observable/observable.pipe.ts

```diff
@@ -0,0 +1,53 @@
+import type {Signal} from '@angular/core';
+import {inject, INJECTOR, untracked} from '@angular/core';
+import {toSignal} from '@angular/core/rxjs-interop';
+import type {Observable} from 'rxjs';
+import {Subject, takeUntil} from 'rxjs';
+
+/**
+ * Abstract class to provide functionality for converting
+ * RxJS Observables to Angular Signals.
+ *
+ * This class is intended to be extended by concrete implementations that define
+ * the `getSignal` method, which returns a `Signal` factory function. This factory
+ * function should be pure and must generate a `Signal` based on the input value and arguments.
+ *
+ * @example
+ * ```
+ * @Pipe({name: 'countdown', pure: false})
+ * export class ToStringPipe extends TuiObservablePipe {
+ *   public transform(value: number): string {
+ *      return this.getSignal(value)();
+ *   }
+ *
+ *   @tuiPure
+ *   protected getSignal(value: number): Signal<R> {
+ *      const countdown$ = timer(0, 1000).pipe(
+ *         map(i => value - i)
+ *         takeWhile(i => i >= 0, true),
+ *      );
+ *
+ *      return this.toSignal(countdown$, value);
+ *   }
+ * }
+ * ```
+ */
+// eslint-disable-next-line @taiga-ui/experience/no-private-esnext-fields
```

Comment:
I believe it is necessary to use ECMA private fields to prevent any conflicts with subclass definitions.

Reply:
unfortunately we can't use private fields now because Safari 13.1 doesn't support it

Reply:
It doesn't matter because TypeScript automatically added the necessary polyfills.

Reply:
Source:
```
class Foo {
  #bar = 123;

  method() {
    console.log(this.#bar);
  }
}
```

Compiled:
```
"use strict";
var __classPrivateFieldGet = (this && this.__classPrivateFieldGet) || function (receiver, state, kind, f) {
    if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a getter");
    if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot read private member from an object whose class did not declare it");
    return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
};
var _Foo_bar;
class Foo {
    constructor() {
        _Foo_bar.set(this, 123);
    }
    method() {
        console.log(__classPrivateFieldGet(this, _Foo_bar, "f"));
    }
}
_Foo_bar = new WeakMap();
```

Reply:
We are just 1 Safari major release away from real private fields. I think we can avoid them for now, we lived without them all this time.